extra/src/main/java/aima/extra/probability/BigDecimalProbabilityNumber.java

```diff
@@ -0,0 +1,381 @@
+package aima.extra.probability;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.math.MathContext;
+import java.math.RoundingMode;
+
+/**
+ * Implementation based on the underlying BigDecimal datatype. Supports higher
+ * precision than DoubleProbabilityNumber (DECIMAL128 precision by default, but
+ * can go upto UNLIMITED). The class is immutable.
+ * 
+ * @author Nagaraj Poti
+ *
+ */
+public class BigDecimalProbabilityNumber implements ProbabilityNumber, Comparable<ProbabilityNumber> {
+
```

Comment:
The ProbabilityNumber interface already extends Comparable<ProbabilityNumber> so you shouldn't have to add to this implements lists as well (applies to other implementations as well).